src/QsCompiler/CompilerOptimizations/CallableInlining.fs

```diff
@@ -81,12 +81,30 @@ type private InliningInfo = {
     body: QsScope
 }
 
+/// A scope transformation that substitutes type parameters according to the given dictionary
+type private ReplaceTypeParams(typeParams: ImmutableDictionary<QsTypeParameter, ResolvedType>) =
+    inherit ScopeTransformation()
+
+    override __.Expression = { new ExpressionTransformation() with
+        override __.Type = { new ExpressionTypeTransformation() with
+            override this.onTypeParameter tp =
+                if typeParams.ContainsKey tp then
+                    typeParams.[tp].Resolution
+                else
+                    let origin = tp.Origin
+                    let name = tp.TypeName
+                    let range = this.onRangeInformation tp.Range
```

Comment:
Are you explicitly rebuilding the type parameter because you need the onRangeInformation for something?

Reply:
Fixed in latest commit